### PR TITLE
fix network local ratelimit connection close without reason

### PR DESCRIPTION
**Pull request by StarryVae**
Commit Message: fix network local ratelimit connection close without reason
Additional Description: N/A
Risk Level: low
Testing: ut
Docs Changes: N/A
Release Notes: N/A

### CODE REVIEW

**Comment by StarryVae** [comment]
same issue like #26856

**Comment by StarryVae** [comment]
@wbpcode PTAL, thanks.

File: test/extensions/filters/network/local_ratelimit/local_ratelimit_test.cc [hunk 1]
```diff
@@ -94,7 +94,7 @@ stat_prefix: local_rate_limit_stats
   ActiveFilter active_filter2(config_);
   EXPECT_CALL(active_filter2.read_filter_callbacks_.connection_.stream_info_,
               setResponseFlag(StreamInfo::ResponseFlag::UpstreamRetryLimitExceeded));
-  EXPECT_CALL(active_filter2.read_filter_callbacks_.connection_, close(_));
+  EXPECT_CALL(active_filter2.read_filter_callbacks_.connection_, close(_, _));
```
Comment (wbpcode) [inline test/extensions/filters/network/local_ratelimit/local_ratelimit_test.cc:97]: Prefer match `local_ratelimit_close_over_limit` here. Thanks.

Comment (StarryVae) [reply test/extensions/filters/network/local_ratelimit/local_ratelimit_test.cc]: done, thanks

**Review by wbpcode — APPROVED** [review]
LGTM. Thanks.

**Pull request merged**